lib/iaas/AzureClient.js

```diff
@@ -154,7 +154,10 @@ class AzureClient extends BaseCloudClient {
           .snapshots
           .deleteMethodAsync(this.settings.resource_group, snapshotName)
           .then(retval => logger.info(`Deleted snapshot ${snapshotName}`, retval))
-          .catch(err => logger.error(`Error occured while deleting snapshot ${snapshotName}`, err));
+          .catch(err => {
+            logger.error('Error occured while deleting snapshot', err);
+            throw err;
```

Comment:
here it would be better if we re-throw the error only if the err is not `Snapshot not found` - Need to check the exact error description when snapshot is not found ? Ideally this would not be the case .

Reply:
@kamath-prasad  the exact error is InvalidSnapshot.NotFound and is aws-sdk specific error.
we here catch for all errors , errors which we are not still aware of .
@jagadish-kb wanted to log the error. Easier to search the log with known string. Not necessary ?

Reply:
@muralisc  OK